packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -47,20 +47,91 @@ def test_metrics_masters_prom(dcos_api_session, prometheus_port):
         assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
 
 
+@retrying.retry(wait_fixed=2000, stop_max_delay=30000)
+def get_metrics_prom(dcos_api_session, prometheus_port, node, expected_metrics):
+    """Assert that expected metrics are present on prometheus port on node.
+
+    Retries on non-200 status or missing expected metrics
+    for up to 30 seconds.
+
+    """
+    response = dcos_api_session.session.request(
+        'GET', 'http://' + node + ':{}/metrics'.format(prometheus_port))
+    assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+    for metric_name in expected_metrics:
+        assert metric_name in response.text
```

Comment:
Looks like there's a Python package called `prometheus_client` that we could use to parse the Prometheus-formatted response: https://github.com/prometheus/client_python/#parser That would allow us to assert that the exact metric key is present, rather than searching for a substring in the response body. But we're using long and specific metric names, so I doubt this'll pass by coincidence. I'm cool with keeping this, but maybe we should file a low-priority ticket to improve this by parsing the response.